test/integration/client_test.go

```diff
@@ -132,3 +145,214 @@ func TestClient(t *testing.T) {
 		}
 	}
 }
+
+func TestMultiWatch(t *testing.T) {
+	// Disable this test as long as it demonstrates a problem.
```

Comment:
nit: I would rather have a TODO: enable the test once the problem is resolved. Up to you on this.